app/scripts/views/mixins/password-strength-mixin.js

```diff
@@ -0,0 +1,53 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'lib/promise',
+  'lib/runtime-require'
+], function (p, runtimeRequire) {
+  'use strict';
+
+  var PasswordStrengthMixin = {
+    _getPasswordStrengthChecker: function () {
+      // returns a promise that resolves once the library is loaded.
+      var self = this;
+      if (! self._passwordStrengthCheckerPromise) {
+        self._passwordStrengthCheckerPromise = runtimeRequire('passwordcheck')
+          .then(function (PasswordCheck) {
+            return new PasswordCheck();
+          });
+      }
+      return self._passwordStrengthCheckerPromise;
+    },
+    /**
+     * Check the password strength. Returns a promise that resolves
+     * when the check is complete.
+     *
+     * Usage:
+     *
+     *   view.checkPasswordStrength(password)
+     *     .then(function (status) {
+     *      // do something with the status
+     *     });
+     *
+     * @method checkPasswordStrength
+     * @returns {Promise}
+     */
+    checkPasswordStrength: function (password) {
+      var self = this;
+      return this._getPasswordStrengthChecker()
+        .then(function (passwordStrengthChecker) {
+          var deferred = p.defer();
```

Comment:
The password strength checker is more a general solution than just a bloom filter. I have a PR to rename `ALL_NUMBERS_LETTERS` => `NOT_STRONG_ENOUGH` to better convey its intent.

The sequence of checks is:
1. Has a checker been passed a password? If not, the status is MISSING_PASSWORD.
1. Is the type of the password a string? If not, the status is PASSWORD_NOT_A_STRING.
1. If the password length is fewer than the required length, the status is PASSWORD_TOO_SHORT.
1. If the password is long enough, check to see if it is all letters or numbers AND fewer characters than minimum password length + 4. If so, NOT_STRONG_ENOUGH.
1. Finally, if the password has met all of those criteria, check the bloomfilter.

The reason for the NOT_STRONG_ENOUGH check is that it enables the bloomfilter data file to be signifcantly smaller. If we outright ban all passwords that contain only letters and numbers and are fewer characters than minlength + 4, then the bloomfilter's dataset can exclude all words between 8 and 12 characters (in our case).